binding/react/src/use_porcupine.ts

```diff
@@ -9,209 +9,133 @@
+  ) => Promise<void>,
+  start: () => Promise<void>,
+  stop: () => Promise<void>,
+  release: () => Promise<void>,
+} => {
+  const porcupineRef = useRef<PorcupineWorker | null>(null);
+  const [wakeWordDetection, setWakeWordDetection] = useState<PorcupineDetection | null>(null);
 
-export function usePorcupine(
-  /** The language-specific worker factory, imported as { PorcupineWorkerFactory }
-   * from the @picovoice/porcupine-web-xx-worker series of packages, where xx is the two-letter language code. */
-  porcupineWorkerFactory: PorcupineWorkerFactory | null,
-  /** usePorcupine Hook Parameters. */
-  porcupineHookArgs: PorcupineHookArgs | null,
-  /** User-defined callback function invoked upon detection of the keywords. */
-  keywordEventHandler: (label: string) => void
-): {
-  /** A state indicating whether the engine is initialized successfully */
-  isLoaded: boolean;
-  /** A state indicating whether the webVoiceProcessor is passing audio to the engine. */
-  isListening: boolean;
-  /** A state indicating whether the Hook returned an error. */
-  isError: boolean | null;
-  /** A string expression of the error. */
-  errorMessage: string | null;
-  /** A pointer to the internal webVoiceProcessor object. */
-  webVoiceProcessor: WebVoiceProcessor | null;
-  /** A method to start listening to the mic and processing the audio. */
-  start: () => void;
-  /** A method to stop listening to the mic. */
-  stop: () => void;
-  /** A method to stop processing the audio. */
-  pause: () => void;
-  /** Setter for keywordEventHandler */
-  setKeywordEventHandler: React.Dispatch<
-    React.SetStateAction<(label: string) => void>
-  >;
-} {
   const [isLoaded, setIsLoaded] = useState(false);
   const [isListening, setIsListening] = useState(false);
-  const [
-    webVoiceProcessor,
-    setWebVoiceProcessor,
-  ] = useState<WebVoiceProcessor | null>(null);
-  const [isError, setIsError] = useState<boolean | null>(null);
-  const [errorMessage, setErrorMessage] = useState<string | null>(null);
-  const [keywordCallback, setKeywordCallback] = useState(
-    () => keywordEventHandler
-  );
-
-  const start = (): boolean => {
-    if (webVoiceProcessor !== null) {
-      webVoiceProcessor.start().then(() => {
-        setIsListening(true);
-        return true;
-      });
+  const [error, setError] = useState<string | null>(null);
+
+  const keywordDetectionCallback = useCallback((keyword: PorcupineDetection) => {
+    setWakeWordDetection(keyword);
+  }, []);
+
+  const errorCallback = useCallback((e: string) => {
+    setError(e);
+  }, []);
+
+  const init = useCallback(async (
+    accessKey: string,
+    keywords: Array<PorcupineKeyword | BuiltInKeyword> | PorcupineKeyword | BuiltInKeyword,
+    model: PorcupineModel,
+    options: PorcupineOptions = {},
+  ): Promise<void> => {
+    if (options.processErrorCallback) {
+      console.warn(`'processErrorCallback' options is not supported, use 'error' state to handle errors.`);
```

Comment:
I used a slightly different message, we should probably use the same one. I wrote:
```
"'processErrorCallback' is only supported in the Picovoice Web SDK. Use the 'error' state to monitor for errors in the React SDK."
```